### PR TITLE
Added Unit Game SS-450

### PR DESCRIPTION
**Pull request by nikhil-chimple**

### CODE REVIEW

File: lib/jamaica/games/unit_game.dart [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+  final OnGameOver onGameOver;
+
+  const UnitGame({
+    Key key,
+    this.question,
+    this.onGameOver,
+  }) : super(key: key);
+
+  @override
+  _UnitGameGameState createState() => _UnitGameGameState();
```
Comment (talapadi) [inline lib/jamaica/games/unit_game.dart]: why is this named UnitGameGameState

File: lib/jamaica/games/unit_game.dart [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+      this.reaction = Reaction.success,
+      this.currentValue = 0,
+      this.success = false});
+  @override
+  String toString() =>
+      '_ChoiceDetail(choice: $choice,  position:$position,currentValue:$currentValue,success:$success , reaction: $reaction)';
+}
+
+class UnitGame extends StatefulWidget {
+  final Map<String, int> question;
```
Comment (talapadi) [inline lib/jamaica/games/unit_game.dart]: Take one number as the input. You can split it internally into unit, ten, hundred

File: lib/jamaica/games/unit_game.dart [hunk 3]
```diff
@@ -33,19 +33,24 @@ class UnitGame extends StatefulWidget {
...
   List<_ChoiceDetail> answerDetails = [];
+  Map<String, int> data = {};
   int score = 5;
   String displayText = '';
 
   @override
   void initState() {
     super.initState();
-    widget.question.forEach((c, d) {
+    data['H'] = widget.question ~/ 100;
```
Comment (talapadi) [inline lib/jamaica/games/unit_game.dart]: what will happen if i send 32 or 8

File: lib/jamaica/games/unit_game.dart [hunk 4]
```diff
@@ -39,22 +39,38 @@ class UnitGame extends StatefulWidget {
...
   void initState() {
     super.initState();
-    data['H'] = widget.question ~/ 100;
-    data['T'] = (widget.question ~/ 10) % 10;
-    data['U'] = widget.question % 10;
+    displayText = widget.question.toString();
+    int i = 0;
+    int j = displayText.length - 1;
+    String position = '';
+    while (i < 3) {
```
Comment (talapadi) [inline lib/jamaica/games/unit_game.dart]: i is always 0. how does this logic work?

Comment (nikhil-chimple) [reply lib/jamaica/games/unit_game.dart]: i am incrementing i down. Here i am trying to add digits in order starting from Unit place

**Pull request merged**